### ISSUE TITLE
Mistake in Chapter 8, fixing dividedBy?

### ISSUE THREAD

**Issue by peterstorm** [larrybotha/haskell-book#1]
Hello!

First of all, thank you for your amazing work on compiling this stuff for Haskellbook! I use it to check my own work, or peak when I get stuck :)

I just wondered if there isn't a mistake in your function where we fix `dividedBy`?

When I test it out, it doesn't handle negative numerator or denominator, because when you `negate` either the numerator or denominator, you can't do your `if else` statement, to flip the number to negative, as it's not negative anymore.

I did a try on a fix, I think maybe works:

```haskell
dividedBy' :: Integer -> Integer -> (DividedByResult, Integer)
dividedBy' num denom = go num denom 0 1
  where go num denom count neg
          | denom == 0  = (DividedByZero, 0)
          | num < 0     = go (negate num) denom count (neg * (-1))
          | denom < 0   = go num (negate denom) count (neg * (-1))
          | num < denom = (Result (count
                            * (if neg == (-1) then (-1) else 1)
                                  ), num)
          | otherwise   = go (num - denom) denom (count + 1) neg
```

**Comment by larrybotha** [comment]
Haha! Thanks for the kind words - I'm basically just standing on the shoulders of giants - when I get stuck I'm learning from the folks I've linked to in the readme :)

I've actually been a little demotivated lately as the chapter on Traversible was killing new

**Comment by peterstorm** [comment]
@larrybotha Awesome, and thank you once again :) Good to hear you're gonna keep going! :D

**Comment by larrybotha** [comment]
Yup. so my implementation is definitely wrong! I've added a couple tests, and the highlighted line is what's important here regarding the negative values: https://github.com/larrybotha/haskell-book/blob/master/chapter-08/dividedByFixed.hs#L31

I'll take a look at your implementation again in the coming days and fix things up :)
